### DISCUSSION TITLE
SSO/OpenID

### DISCUSSION THREAD

**Discussion by igbjnI05bF** [Stirling-Tools/Stirling-PDF#781] [Ideas]
I just discovered this tool and it seems fantastic. I have made my instance publicly accessible over the Internet so I do not want to disable authentication entirely, but I'd also not like to give everybody in my small company yet another account username/password. SSO integration using OpenID would be very appreciated!

**Comment by sahilph** [comment]
For SSO, Please track the progress here: https://github.com/Stirling-Tools/Stirling-PDF/pull/1140

**Reply by igbjnI05bF** [reply]
So glad somebody is working on this, can't wait for it to be released! Thanks!

**Comment by colfes** [comment]
Is there a way to disable the "normal" Login field, when using OIDC? I have an instance running with only one OIDC provider, and the user experience is not perfect yet.

Is there a way I haven't found to
- disable username/password login entirely
- already show a customized text instead of the "Login with Single Sign-on" at the Login page? (We always use "Login with _CompanyName_", users might not know what Single Sign-On is.)
- already login on klick on the first login button, avoiding to select a SSO provider when there is only one anyway?

Help is greatly appreciated!

**Reply by Frooodle** [reply]
There is currently no inbuilt functionality for this 

Our SSO is heavily under development and additionally some more advanced SSO features may only be added to our business model if it is mostly used in larger companies etc

**Reply by colfes** [reply]
Thanks for your response! My "company" is the student body of a german university, so we would likely not be able to buy a license. If these features will be implemented, consider an "education" version or similar. We might just use the version without login in the meantime, we'll see. (Currently, there is also no option to add an Imprint & Privacy Policy which is legally required in the EU, but I will open a seperate issue about that)

**Reply by Frooodle** [reply]
We do have options to add privacy policy etc 
Please check your settings.yml under legal!
